src/awkward/_behavior.py

```diff
@@ -100,29 +100,32 @@ def find_ufunc_generic(
 
 
 def find_ufunc(behavior: Mapping | None, signature: tuple) -> UfuncLike | None:
-    if all(s is not None for s in signature):
-        behavior = overlay_behavior(behavior)
+    if any(s is None for s in signature):
+        return None
+
+    behavior = overlay_behavior(behavior)
 
-        # Special case all strings or hashable types.
+    # Try and fast-path the lookup
+    try:
+        return behavior[signature]
+    except KeyError:
+        # We didn't find an exact overload, and we won't find any!
         if all(isinstance(x, str) for x in signature):
-            return behavior.get(signature)
-        else:
-            for key, custom in behavior.items():
-                if (
-                    isinstance(key, tuple)
-                    and len(key) == len(signature)
-                    and key[0] == signature[0]
-                    and all(
-                        k == s
-                        or (
-                            isinstance(k, type)
-                            and isinstance(s, type)
-                            and issubclass(s, k)
-                        )
-                        for k, s in zip(key[1:], signature[1:])
-                    )
-                ):
-                    return custom
+            return None
```

Comment:
If a signature (apart from the ufunc itself—or does the signature-part start _after_ the ufunc?) is all strings and we didn't find it with a hash-lookup, we won't find it by iterating over `behavior.keys()`, so don't even bother.

Good! :heavy_check_mark: